sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/ChatAsyncClient.java

```diff
@@ -36,11 +36,11 @@
     private final ClientLogger logger = new ClientLogger(ChatAsyncClient.class);
 
     private final AzureCommunicationChatServiceImpl chatServiceClient;
-    private final ChatImpl chatClient;
+    private final ChatsImpl chatClient;
```

Comment:
Autorest generates it like that. I pinged the AR team and they say it can't be configured to be non-plural and seem to have strong opinions that it should be plural. We were manually renaming it after every swagger change but this was becoming tedious for something that is internal only.

Reply:
There are ways to rename classes with autorest either via annotations or with transforms, so we would not need to do it manually every time a new swagger comes in . Using Chats (or the function we have below call getChats()) where the expected object is singular is going to be confusing. @JianpingChen do you know how to do this in JAVA ?

Reply:
The only places that consume this directly are internal and all auto-generated.

Reply:
ok, if you reached out to autorest folks and there is no way around this I guess we will need to live with it. 
This is an open source package, so even if this is consumed just internally, calling something .getChats where a chatclient is expected is quite unintuitive. I would argue it is even better to do it manually for now, and open a defect with the autorest team.

Reply:
https://github.com/Azure/autorest.java/issues/903